dbms/src/Dictionaries/ClickHouseDictionarySource.cpp

```diff
@@ -143,4 +159,19 @@ BlockInputStreamPtr ClickHouseDictionarySource::createStreamForSelectiveLoad(con
     return std::make_shared<RemoteBlockInputStream>(pool, query, sample_block, context);
 }
 
+std::string ClickHouseDictionarySource::doInvalidateQuery(const std::string & request) const
+{
+    Block invalidate_sample_block;
```

Comment:
This block is `sample_block`, which is used as header. It must contain single string column. Look at similar dictionaries: https://github.com/yandex/ClickHouse/blob/master/dbms/src/Dictionaries/MySQLDictionarySource.cpp#L219-L221 and https://github.com/yandex/ClickHouse/blob/master/dbms/src/Dictionaries/ODBCDictionarySource.cpp#L187-L189

Reply:
No! It does not work with CH.

Reply:
ok